### ISSUE TITLE
LinkedInResourceOwner inconsistency with some other resource owners

### ISSUE THREAD

**Issue by norkunas** [thephpleague/oauth2-linkedin#4]
LinkedIn resource owner has these methods named like this:

``` php
public function getFirstname();
public function getLastname();
```

While facebook and google providers has these:

``` php
public function getFirstName();
public function getLastName();
```

Is this expected naming?

**Comment by stevenmaguire** [comment]
That's a great question. I believe it is the way it is from the previous iteration of this project. I am ok with changing the method names to follow "convention" here and it is worth noting that it would force the package to a major version bump as that change would be a breaking one.

Can you share a bit more as to why this question has come up? Are you hoping to use the same code to access resource owner results for multiple providers? Is this simply the result of noting the difference and feeling like it should be corrected?

**Comment by norkunas** [comment]
Yes, my point was about to use same code across different providers which supports firstName and lastName and i think it should be corrected. Also I think we could annotate the class with @method so it would be backwards compatible and IDE would recognize it.

**Comment by shadowhand** [comment]
Does PHP care if the method names are referred to exactly as capitalized? I thought PHP was case insensitive with method names but not variables?

**Comment by stevenmaguire** [comment]
Good catch @shadowhand, thank you for chiming in here. I completely omitted that information from my brain when writing my response; been writing too much Ruby lately...

The project has now been updated to change the method names to follow "convention" and a new patch release [0.4.1](https://github.com/thephpleague/oauth2-linkedin/releases/tag/0.4.1) has been cut. I've updated the tests to [demonstrate that the case of the methods do not matter](https://github.com/thephpleague/oauth2-linkedin/blob/master/test/src/Provider/LinkedInTest.php#L120). 

@norkunas, you shouldn't _need_ a new version to make your code work, so long as all the resource owners you are working with have support for `getfirstname` and `getlastname`.
